modules/failure-handler/src/main/java/org/apache/ignite/internal/failure/FailureProcessor.java

```diff
@@ -49,6 +51,8 @@ public class FailureProcessor implements IgniteComponent {
     /** Node name. */
     private final String nodeName;
 
+    private volatile @Nullable FailureHandler interceptor;
```

Comment:
it will be used with different threads, so it required have volatile here or synchronized settet, which you suggested below. In this case it doesn't matter which approach to use.

Reply:
it`s matter, check: if you use volatile approach - check can passed but in next line you can obtain NPE
        if (interceptor != null) {
            interceptor.onFailure(nodeName, failureCtx);
        }

Reply:
agree, thanks